### PR TITLE
MINOR: fixes lgtm.com warnings

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
\cc @bbejeck @sebastiansemmle 

Cf. https://lgtm.com/projects/g/apache/kafka/alerts/?mode=list

**Review by bbejeck — APPROVED** [review]
Thanks, @mjsax LGTM

**Comment by bbejeck** [comment]
retest this please

**Comment by ghost** [comment]
Thanks @mjsax ! The changes LGTM as well.

@bbejeck @mjsax  If you like the checks you could enable our Pull Request integration, where we automatically run our analysis on every created PR as a check. If we find anything you can then browse through the results. The analysis runs on our infrastructure and only requires a single click to enable (the owner of your GitHub mirror will have to [do it](https://lgtm.com/help/lgtm/using-lgtm-analysis-continuous-integration)).

For an example of what this looks like, you can go over to the `numpy` project which recently enabled our Python analysis: https://github.com/numpy/numpy/pull/10630  (you need to be logged into GitHub to view it).

Let me know if you are interested and I point you in the right direction.

**Comment by mjsax** [comment]
@SebastianBauersfeld @sebastiansemmle (which one is the right one?)

Thanks for pointing out that we could integrate those check. We cannot make this decision directly -- as an Apache project, there are some guidelines to follow (also not sure, if we need help from Apache Infra team in case we decide that we want this integration).

**Comment by mjsax** [comment]
Retest this please

**Comment by zbazztian** [comment]
@mjsax Understood.

It is `@sebastiansemmle`, I logged in with the wrong one.

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java [hunk 1]
```diff
@@ -385,11 +385,16 @@ public void writeAsText(final String filePath,
             throw new TopologyException("filePath can't be an empty string");
         }
         String name = builder.newProcessorName(PRINTING_NAME);
+        PrintWriter printWriter = null;
         try {
-            PrintWriter printWriter = new PrintWriter(filePath, StandardCharsets.UTF_8.name());
+            printWriter = new PrintWriter(filePath, StandardCharsets.UTF_8.name());
             builder.internalTopologyBuilder.addProcessor(name, new KStreamPrint<>(new PrintForeachAction(printWriter, defaultKeyValueMapper, label)), this.name);
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java]: @bbejeck @sebastiansemmle Was just thinking once more about this change. And it seem that the original code is actually correct. We pass in the `PrintWriter` into the builder and use it later on. If we close it here in `finally` we cannot use the `PrintWriter` later... And `PrintWriter` is closed correctly within `PrintForeachAction#close()`.

Comment (zbazztian) [reply streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java]: You are right, indeed. It is supposed to be closed at a later point in time.

Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java]: Updated the PR to clean this up.

**Comment by mjsax** [comment]
Updated this PR including a cleanup to `Printed` and how we handle STDOUT... Let me know what you think. We had some discussions about this, when introducing `Printed` -- thus, I am not sure if I might have missed something.

File: clients/src/main/java/org/apache/kafka/common/cache/LRUCache.java [hunk 2]
```diff
@@ -29,7 +29,7 @@ public LRUCache(final int maxSize) {
         cache = new LinkedHashMap<K, V>(16, .75f, true) {
             @Override
             protected boolean removeEldestEntry(Map.Entry<K, V> eldest) {
-                return size() > maxSize;
+                return this.size() > maxSize; // require this. prefix to make lgtm.com happy
```
Comment (guozhangwang) [inline clients/src/main/java/org/apache/kafka/common/cache/LRUCache.java:32]: nit: this is not only to make lgtm happy right? We can at least save one virtual call in java byte code (assuming JIT is not yet that smart :P)

**Review by guozhangwang — APPROVED** [review]
LGTM!

**Comment by mjsax** [comment]
Retest this please

**Comment by mjsax** [comment]
Retest this please

**Review by dguy — APPROVED** [review]
Thanks @mjsax, LGTM

**Comment by mjsax** [comment]
Retest this please.

**Comment by mjsax** [comment]
Rebased (hope to get a green build than).
Also added one more commit with minor code cleanup.

File: clients/src/main/java/org/apache/kafka/common/cache/LRUCache.java (hunk 2 above)
Comment (ijuma) [reply clients/src/main/java/org/apache/kafka/common/cache/LRUCache.java:32]: How is this saving a virtual call?

**Pull request merged**